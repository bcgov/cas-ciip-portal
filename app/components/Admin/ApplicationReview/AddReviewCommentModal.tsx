import React, {useState} from 'react';
import {Modal, Button, Form, FormCheck} from 'react-bootstrap';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faLock, faEye} from '@fortawesome/free-solid-svg-icons';

interface Props {
  show: boolean;
  title: string;
  onSubmit: ({commentText: string, isInternalComment: boolean}) => void;
  onHide: () => void;
}

export const AddReviewCommentModal: React.FunctionComponent<Props> = ({
  show,
  title,
  onSubmit,
  onHide
}) => {
  const [isInternalComment, setIsInternalComment] = useState(false);
  const [commentText, setCommentText] = useState('');

  const clearForm = () => {
    setIsInternalComment(false);
    setCommentText('');
  };
  const handleHide = () => {
    onHide();
    clearForm();
  };

  return (
    <Modal
      id="add-comment-modal"
      size="lg"
      show={show}
      onHide={handleHide}
      aria-labelledby="add-comment-title"
    >
      <Modal.Header closeButton>
        <h2 id="add-comment-title">{title}</h2>
      </Modal.Header>
      <Modal.Body>
        <Form.Group controlId="comment">
          <Form.Label>Comment:</Form.Label>
          <Form.Control
            as="textarea"
            rows={3}
            value={commentText}
            onChange={(e) => setCommentText(e.target.value)}
          />
        </Form.Group>
        <FormCheck>
          <FormCheck.Label>
            <FormCheck.Input
              checked={isInternalComment}
              onChange={(e) => setIsInternalComment(e.target.checked)}
            />
            Visible for <strong>internal</strong> use only
          </FormCheck.Label>
        </FormCheck>
      </Modal.Body>
      <Modal.Footer>
        <span>
          This comment will {isInternalComment && <strong>not</strong>} be
          visible to the applicant.
          <FontAwesomeIcon
            id="visibility-icon"
            icon={isInternalComment ? faLock : faEye}
          />
        </span>
        <Button
          id="submit"
          onClick={() => {
            if (commentText.trim().length > 0) {
              onSubmit({commentText, isInternalComment});
              clearForm();
            }
          }}
        >
          Add Comment
        </Button>
      </Modal.Footer>
      <style jsx>{`
        :global(.modal-header) {
          background: #036;
          color: #fff;
        }
        :global(button.close) {
          color: #fff;
        }
        :global(#add-comment-modal .modal-body) {
          padding: 1rem 2rem;
        }
        :global(#add-comment-modal .form-check) {
          margin: 1rem 0 0 0;
        }
        :global(.form-check-label, .form-check-input) {
          cursor: pointer;
        }
        :global(#visibility-icon) {
          margin: 0 1em 0 0.5em;
        }
        h2 {
          margin: 0;
        }
      `}</style>
    </Modal>
  );
};

export default AddReviewCommentModal;
