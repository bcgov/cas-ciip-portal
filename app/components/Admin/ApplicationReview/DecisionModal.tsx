import React from 'react';
import {Modal, Button, Alert} from 'react-bootstrap';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faEnvelope, faCheck, faTimes} from '@fortawesome/free-solid-svg-icons';
import ModalSharedStyles from 'components/ModalSharedStyles';
import {getUserFriendlyStatusLabel} from 'lib/text-transforms';

interface Props {
  show: boolean;
  onHide: () => void;
  onDecision: (decision: string) => void;
  currentStatus: string;
}

export const DecisionModal: React.FunctionComponent<Props> = ({
  show,
  onHide,
  onDecision,
  currentStatus
}) => {
  const decisionHasBeenMade = currentStatus !== 'SUBMITTED';
  const title = decisionHasBeenMade
    ? `Revert Decision`
    : 'Make a Decision or Request Changes';

  const handleDecision = (e) => {
    onDecision(e.target.value);
  };

  const makeDecisionContent = (
    <>
      <Modal.Body>
        <p>
          Once a decision is made or changes are requested, general comments
          will become visible to the applicant, who will be{' '}
          <strong>immediately notified of the decision by email</strong>.
        </p>
        <p className="d-flex justify-content-center">
          <Button
            value="REQUESTED_CHANGES"
            variant="outline-secondary"
            size="lg"
            onClick={handleDecision}
          >
            <FontAwesomeIcon icon={faEnvelope} />
            Request Changes
          </Button>
        </p>
      </Modal.Body>
      <Modal.Footer className="d-flex justify-content-around">
        <Button
          value="APPROVED"
          variant="success"
          size="lg"
          onClick={handleDecision}
        >
          <FontAwesomeIcon icon={faCheck} />
          Approve
        </Button>
        <strong>OR</strong>
        <Button
          value="REJECTED"
          variant="danger"
          size="lg"
          onClick={handleDecision}
        >
          <FontAwesomeIcon icon={faTimes} />
          Reject
        </Button>
      </Modal.Footer>
    </>
  );
  const revertDecisionContent = (
    <>
      <Modal.Body>
        <p>
          The applicant was notified by email of the following decision:{' '}
          <strong>{getUserFriendlyStatusLabel(currentStatus)}</strong>
        </p>
        <p>Would you like to revert this decision?</p>
        <p>
          Doing so will change this application&apos;s status back to{' '}
          <strong>Submitted</strong> and you will have the opportunity to make
          further changes to the review before making a new decision or
          requesting changes.
        </p>
        {currentStatus === 'REQUESTED_CHANGES' && (
          <Alert variant="secondary">
            <strong>Note:</strong> If the applicant begins drafting a new
            revision of the application, it will no longer be possible to revert
            a decision of <strong>Changes requested</strong>.
          </Alert>
        )}
      </Modal.Body>
      <Modal.Footer>
        <Button value="SUBMITTED" onClick={handleDecision}>
          Revert Decision
        </Button>
      </Modal.Footer>
    </>
  );
  return (
    <Modal
      show={show}
      centered
      size="lg"
      onHide={onHide}
      aria-labelledby="heading"
    >
      <Modal.Header closeButton>
        <h2 id="heading">{title}</h2>
      </Modal.Header>
      {!decisionHasBeenMade && makeDecisionContent}
      {decisionHasBeenMade && revertDecisionContent}
      <style jsx>{ModalSharedStyles}</style>
      <style jsx>{`
        :global(.modal-body) {
          padding: 2rem;
        }
        :global(.btn svg.svg-inline--fa) {
          margin-right: 0.5em;
        }
      `}</style>
    </Modal>
  );
};

export default DecisionModal;
