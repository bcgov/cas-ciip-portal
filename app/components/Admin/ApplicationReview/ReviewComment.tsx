import React from "react";
import { Button, OverlayTrigger, Tooltip } from "react-bootstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTrash } from "@fortawesome/free-solid-svg-icons";
import { dateTimeFormat } from "functions/formatDates";

interface Props {
  id: string;
  description: string;
  createdAt: string;
  createdBy: string;
  viewOnly: boolean;
  isResolved: boolean;
  onResolveToggle: (id: string, resolve: boolean) => void;
  onDelete: (id: string) => void;
}

export const ReviewComment: React.FunctionComponent<Props> = ({
  id,
  description,
  createdAt,
  createdBy,
  viewOnly,
  isResolved,
  onResolveToggle,
  onDelete,
}) => {
  return (
    <li>
      <div className="text-block">
        <p>{description}</p>
        <p className="timestamp">
          <span>{createdBy}</span>
          {`${dateTimeFormat(createdAt, "minutes")}`}
        </p>
      </div>
      {!viewOnly && (
        <div>
          <Button
            size="sm"
            variant="outline-primary"
            style={{ marginRight: "0.5em" }}
            onClick={() => onResolveToggle(id, !isResolved)}
          >
            {`${isResolved ? "Unresolve" : "Resolve"}`}
          </Button>
          <OverlayTrigger
            placement="top"
            overlay={(props) => (
              <Tooltip id="delete" {...props}>
                Delete comment
              </Tooltip>
            )}
          >
            <Button
              size="sm"
              aria-label="Delete"
              variant="secondary"
              onClick={() => onDelete(id)}
            >
              <FontAwesomeIcon icon={faTrash} />
            </Button>
          </OverlayTrigger>
        </div>
      )}
      <style jsx>{`
        li {
          list-style-type: none;
          padding: 0;
          margin-bottom: 1rem;
          width: 100%;
          display: inline-flex;
          justify-content: space-between;
        }
        li:last-of-type {
          margin: 0;
        }
        .text-block {
          max-width: calc(100% - 120px - 0.5rem);
        }
        .text-block ~ div {
          margin-left: 0.5rem;
        }
        p {
          margin-bottom: 0.5em;
          line-height: 1.2;
          font-size: 0.9rem;
        }
        .timestamp {
          font-size: 0.8rem;
          margin-bottom: 0;
          line-height: 1.5;
          color: #666;
        }
        .timestamp span {
          margin-right: 0.5em;
        }
      `}</style>
    </li>
  );
};

export default ReviewComment;
