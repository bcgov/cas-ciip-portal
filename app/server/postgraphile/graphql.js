const {
  createPostGraphileSchema,
  withPostGraphileContext,
} = require("postgraphile");
const { graphql } = require("graphql");
const authenticationPgSettings = require("./authenticationPgSettings");
const { pgPool, getDatabaseUrl } = require("../db/databaseConnectionService");
const postgraphileOptions = require("./postgraphileOptions");

const pgSettings = (req) => {
  const opts = {
    ...authenticationPgSettings(req),
  };
  return opts;
};

let postgraphileSchemaSingleton;

const postgraphileSchema = async () => {
  if (!postgraphileSchemaSingleton) {
    postgraphileSchemaSingleton = await createPostGraphileSchema(
      getDatabaseUrl(),
      process.env.DATABASE_SCHEMA || "ggircs_portal",
      postgraphileOptions()
    );
  }

  return postgraphileSchemaSingleton;
};

async function performQuery(query, variables, request) {
  const settings = pgSettings(request);

  return withPostGraphileContext(
    {
      pgPool,
      pgSettings: settings,
    },
    async (context) => {
      // `context` object, which should NOT be used outside of this
      // function.
      const result = graphql(
        await postgraphileSchema(),
        query,
        null,
        { ...context }, // You can add more to context if you like
        variables
      );
      return result;
    }
  );
}

module.exports = {
  performQuery,
};
