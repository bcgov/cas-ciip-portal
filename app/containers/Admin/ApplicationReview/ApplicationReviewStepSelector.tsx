import React from 'react';
import {Row, Button, ListGroup} from 'react-bootstrap';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {
  faCheck,
  faComments,
  faLock,
  faTimes,
  faHourglassHalf,
  faKey,
  faEnvelope
} from '@fortawesome/free-solid-svg-icons';
import {graphql, createFragmentContainer} from 'react-relay';
import {CiipApplicationRevisionStatus} from 'applicationReviewQuery.graphql';
import {ApplicationReviewStepSelector_applicationReviewSteps} from '__generated__/ApplicationReviewStepSelector_applicationReviewSteps.graphql';
import {capitalize} from 'lib/text-transforms';

interface Props {
  applicationReviewSteps: ApplicationReviewStepSelector_applicationReviewSteps;
  selectedStep: string;
  onSelectStep: (stepId: string) => void;
  decisionOrChangeRequestStatus: CiipApplicationRevisionStatus;
  newerDraftExists: boolean;
  onDecisionOrChangeRequestAction: () => void;
  changeDecision?: () => void;
}

const DECISION_BS_VARIANT = {
  SUBMITTED: undefined,
  APPROVED: 'success',
  REJECTED: 'danger',
  REQUESTED_CHANGES: 'secondary'
};

const DECISION_ICON = {
  APPROVED: faCheck,
  REJECTED: faTimes,
  REQUESTED_CHANGES: faHourglassHalf
};

const DECISION_BUTTON_TEXT = {
  APPROVED: 'Application has been approved and applicant notified',
  REJECTED: 'Application has been rejected and applicant notified',
  REQUESTED_CHANGES: 'Changes have been requested and applicant notified'
};

export const ApplicationReviewStepSelector: React.FunctionComponent<Props> = ({
  applicationReviewSteps,
  selectedStep,
  onSelectStep,
  decisionOrChangeRequestStatus,
  newerDraftExists,
  onDecisionOrChangeRequestAction,
  changeDecision
}) => {
  const steps = applicationReviewSteps.edges;
  const allStepsAreComplete = steps.every((edge) => edge.node.isComplete);
  const decisionHasBeenMade = decisionOrChangeRequestStatus !== 'SUBMITTED';
  const decisionOrChangeRequestIsDisabled =
    !allStepsAreComplete || decisionHasBeenMade;
  const renderStepStatusIcon = (icon) => (
    <FontAwesomeIcon
      icon={icon}
      style={{
        position: 'absolute',
        left: '1.2rem',
        top: 'calc(50% - 0.5em)'
      }}
    />
  );
  const changeDecisionButton = (
    <Button variant="link" onClick={changeDecision}>
      <FontAwesomeIcon icon={faKey} style={{marginRight: 8}} />
      Change decision
    </Button>
  );
  const newDraftIndicator = (
    <Button
      variant="link"
      disabled
      style={{opacity: 1, color: '#343a40', textAlign: 'left'}}
    >
      <FontAwesomeIcon icon={faLock} style={{marginRight: 8}} />
      There is a newer draft of this application.
    </Button>
  );
  return (
    <Row>
      <div className="col-xxl-6 col-xl-7 col-lg-8 col-md-10">
        <ListGroup as="ul" id="selector">
          {steps.map((edge) => {
            const {reviewStepId, isComplete} = edge.node;
            const isSelectedStep = edge.node.id === selectedStep;
            const callToAction = `${isComplete ? '' : 'Open'} ${capitalize(
              edge.node.reviewStepByReviewStepId.stepName
            )} review ${isComplete ? 'completed' : ''}`;
            return (
              <ListGroup.Item
                className="review-step-option"
                as="li"
                action
                tabIndex={0}
                key={reviewStepId}
                active={isSelectedStep}
                onClick={() => onSelectStep(edge.node.id)}
                onKeyDown={(e) => {
                  if (e.key === 'Enter') onSelectStep(edge.node.id);
                }}
                style={{
                  cursor: 'pointer',
                  position: 'relative',
                  paddingLeft: '3rem'
                }}
              >
                {renderStepStatusIcon(isComplete ? faCheck : faComments)}
                {callToAction}
              </ListGroup.Item>
            );
          })}
          <ListGroup.Item
            id="open-decision-dialog"
            as="li"
            action
            active={!decisionOrChangeRequestIsDisabled}
            key="decision"
            tabIndex={0}
            variant={DECISION_BS_VARIANT[decisionOrChangeRequestStatus]}
            style={{cursor: 'pointer', paddingLeft: '3rem'}}
            disabled={decisionOrChangeRequestIsDisabled}
            aria-disabled={decisionOrChangeRequestIsDisabled}
            onClick={() => {
              if (!decisionOrChangeRequestIsDisabled)
                onDecisionOrChangeRequestAction();
            }}
            onKeyDown={(e) => {
              if (e.key === 'Enter' && !decisionOrChangeRequestIsDisabled)
                onDecisionOrChangeRequestAction();
            }}
          >
            {decisionHasBeenMade && (
              <FontAwesomeIcon
                icon={DECISION_ICON[decisionOrChangeRequestStatus]}
                style={{
                  position: 'absolute',
                  left: '1.2rem',
                  top: 'calc(50% - 0.5em)'
                }}
              />
            )}
            {decisionHasBeenMade &&
              DECISION_BUTTON_TEXT[decisionOrChangeRequestStatus]}
            {!decisionHasBeenMade &&
              (decisionOrChangeRequestIsDisabled ? (
                <em>
                  Complete the reviews above to make a decision or request
                  changes
                </em>
              ) : (
                <>
                  <FontAwesomeIcon
                    icon={faEnvelope}
                    style={{
                      position: 'absolute',
                      left: '1.2rem',
                      top: 'calc(50% - 0.5em)'
                    }}
                  />
                  Make a decision or request changes
                </>
              ))}
          </ListGroup.Item>
        </ListGroup>
      </div>
      {changeDecision && (
        <div
          id="change-decision"
          className="col-md-12 col-lg-4 col-xl-5 col-xxl-6"
        >
          {newerDraftExists ? newDraftIndicator : changeDecisionButton}
        </div>
      )}
      <style jsx>{`
        #change-decision {
          display: flex;
          align-items: flex-end;
        }
      `}</style>
      <style jsx global>{`
        #selector .list-group-item.active {
          z-index: auto;
          background: #38598a;
        }
        .list-group-item#open-decision-dialog.active {
          background: #036;
        }
        #selector .list-group-item-danger.disabled,
        .list-group-item-danger:disabled {
          color: #721c24;
          background-color: #f5c6cb;
        }
        #selector .list-group-item-success.disabled,
        .list-group-item-success:disabled {
          color: #155724;
          background-color: #c3e6cb;
        }
        #selector .list-group-item-secondary.disabled,
        .list-group-item-secondary:disabled {
          color: #383d41;
          background-color: #d6d8db;
        }
      `}</style>
    </Row>
  );
};

export default createFragmentContainer(ApplicationReviewStepSelector, {
  applicationReviewSteps: graphql`
    fragment ApplicationReviewStepSelector_applicationReviewSteps on ApplicationReviewStepsConnection {
      edges {
        node {
          id
          isComplete
          reviewStepId
          reviewStepByReviewStepId {
            stepName
          }
        }
      }
    }
  `
});
