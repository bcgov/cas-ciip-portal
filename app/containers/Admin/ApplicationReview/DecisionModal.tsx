import React from 'react';
import {graphql, createFragmentContainer, RelayProp} from 'react-relay';
import {DecisionModal_applicationRevisionStatus} from '__generated__/DecisionModal_applicationRevisionStatus.graphql';
import createApplicationRevisionStatusMutation from 'mutations/application/createApplicationRevisionStatusMutation';
import {CiipApplicationRevisionStatus} from 'createApplicationRevisionStatusMutation.graphql';
import {Modal, Button, Alert} from 'react-bootstrap';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faEnvelope, faCheck, faTimes} from '@fortawesome/free-solid-svg-icons';
import ModalSharedStyles from 'components/ModalSharedStyles';
import {getUserFriendlyStatusLabel} from 'lib/text-transforms';

interface Props {
  show: boolean;
  onHide: () => void;
  onDecision: () => void;
  relay: RelayProp;
  applicationRevisionStatus: DecisionModal_applicationRevisionStatus;
}

export const DecisionModal: React.FunctionComponent<Props> = ({
  show,
  onHide,
  onDecision,
  relay,
  applicationRevisionStatus
}) => {
  const {applicationId, versionNumber} = applicationRevisionStatus;
  const status = applicationRevisionStatus.applicationRevisionStatus;
  const decisionHasBeenMade = status !== 'SUBMITTED';
  const title = decisionHasBeenMade
    ? `Revert Decision`
    : 'Make a Decision or Request Changes';

  const saveDecision = async (decision) => {
    onDecision();
    const variables = {
      input: {
        applicationRevisionStatus: {
          applicationId,
          applicationRevisionStatus: decision as CiipApplicationRevisionStatus,
          versionNumber
        }
      }
    };
    await createApplicationRevisionStatusMutation(relay.environment, variables);
  };
  const makeDecisionContent = (
    <>
      <Modal.Body>
        <p>
          Once a decision is made or changes are requested, general comments
          will become visible to the applicant, who will be{' '}
          <strong>immediately notified of the decision by email</strong>.
        </p>
        <p className="d-flex justify-content-center">
          <Button
            variant="outline-secondary"
            size="lg"
            onClick={() => saveDecision('REQUESTED_CHANGES')}
          >
            <FontAwesomeIcon icon={faEnvelope} style={{marginRight: '0.5em'}} />
            Request Changes
          </Button>
        </p>
      </Modal.Body>
      <Modal.Footer className="d-flex justify-content-around">
        <Button
          variant="success"
          size="lg"
          onClick={() => saveDecision('APPROVED')}
        >
          <FontAwesomeIcon icon={faCheck} />
          Approve
        </Button>
        <strong>OR</strong>
        <Button
          variant="danger"
          size="lg"
          onClick={() => saveDecision('REJECTED')}
        >
          <FontAwesomeIcon icon={faTimes} />
          Reject
        </Button>
      </Modal.Footer>
    </>
  );
  const revertDecisionContent = (
    <>
      <Modal.Body>
        <p>
          The applicant was notified by email of the following decision:{' '}
          <strong>{getUserFriendlyStatusLabel(status)}</strong>
        </p>
        <p>Would you like to revert this decision?</p>
        <p>
          Doing so will change this application&apos;s status back to{' '}
          <strong>Submitted</strong> and you will have the opportunity to make
          further changes to the review before making a new decision or
          requesting changes.
        </p>
        {status === 'REQUESTED_CHANGES' && (
          <Alert variant="secondary">
            <strong>Note:</strong> If the applicant begins drafting a new
            revision of the application, it will no longer be possible to revert
            a decision of <strong>Changes requested</strong>.
          </Alert>
        )}
      </Modal.Body>
      <Modal.Footer>
        <Button onClick={() => saveDecision('SUBMITTED')}>
          Revert Decision
        </Button>
      </Modal.Footer>
    </>
  );
  return (
    <Modal show={show} centered size="lg" onHide={onHide}>
      <Modal.Header closeButton>
        <h2>{title}</h2>
      </Modal.Header>
      {!decisionHasBeenMade && makeDecisionContent}
      {decisionHasBeenMade && revertDecisionContent}
      <style jsx>{ModalSharedStyles}</style>
      <style jsx>{`
        :global(.modal-body) {
          padding: 2rem;
        }
        :global(.btn svg.svg-inline--fa) {
          margin-right: 0.5em;
        }
      `}</style>
    </Modal>
  );
};

export default createFragmentContainer(DecisionModal, {
  applicationRevisionStatus: graphql`
    fragment DecisionModal_applicationRevisionStatus on ApplicationRevisionStatus {
      applicationRevisionStatus
      applicationId
      versionNumber
    }
  `
});
