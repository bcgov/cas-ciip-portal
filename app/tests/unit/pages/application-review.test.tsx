import React from 'react';
import {shallow} from 'enzyme';
import ApplicationReview from 'pages/application-review';
import {applicationReviewQueryResponse} from 'applicationReviewQuery.graphql';

describe('The application-review page', () => {
  const query: applicationReviewQueryResponse['query'] = {
    session: {
      ' $fragmentRefs': {
        defaultLayout_session: true
      }
    },
    application: {
      ' $fragmentRefs': {
        ApplicationDetailsContainer_application: true
      },
      rowId: 1,
      applicationRevisionStatus: {
        ' $fragmentRefs': {
          ApplicationRevisionStatusContainer_applicationRevisionStatus: true
        }
      }
    },
    applicationRevision: {
      ' $fragmentRefs': {
        IncentiveCalculatorContainer_applicationRevision: true
      }
    },
    ' $fragmentRefs': {
      ApplicationCommentsContainer_query: true,
      ApplicationDetailsContainer_query: true
    }
  };

  it('It matches the last accepted Snapshot', () => {
    const wrapper = shallow(<ApplicationReview router={null} query={query} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('It passes the applicationRevision prop to the IncentiveCalculator', () => {
    const wrapper = shallow(<ApplicationReview router={null} query={query} />);
    expect(
      wrapper
        .find('Relay(IncentiveCalculator)')
        .first()
        .prop('applicationRevision')
    ).toBe(query.applicationRevision);
  });
});
