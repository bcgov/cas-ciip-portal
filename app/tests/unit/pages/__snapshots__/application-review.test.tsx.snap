// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The application-review page It matches the last accepted Snapshot 1`] = `
<Relay(DefaultLayout)
  allowedGroups={
    Array [
      "Incentive Analyst",
    ]
  }
  session={
    Object {
      " $fragmentRefs": Object {
        "defaultLayout_session": true,
      },
    }
  }
  width="wide"
>
  <Relay(ApplicationRevisionStatusComponent)
    applicationRevisionStatus={
      Object {
        " $fragmentRefs": Object {
          "ApplicationRevisionStatusContainer_applicationRevisionStatus": true,
        },
      }
    }
    applicationRowId={1}
  />
  <hr
    className="jsx-1070550671"
  />
  <ForwardRef
    className="application-container"
    noGutters={false}
  >
    <Col
      className="application-body"
      md={8}
    >
      <Relay(ApplicationDetailsComponent)
        application={
          Object {
            " $fragmentRefs": Object {
              "ApplicationDetailsContainer_application": true,
            },
            "applicationRevisionStatus": Object {
              " $fragmentRefs": Object {
                "ApplicationRevisionStatusContainer_applicationRevisionStatus": true,
              },
            },
            "orderedFormResults": Object {
              "edges": Array [
                Object {
                  "node": Object {
                    " $fragmentRefs": Object {
                      "ApplicationCommentsContainer_formResult": true,
                    },
                    "id": "abc",
                  },
                },
              ],
            },
            "rowId": 1,
          }
        }
        query={
          Object {
            " $fragmentRefs": Object {
              "ApplicationDetailsContainer_query": true,
              "IncentiveCalculatorContainer_query": true,
            },
            "application": Object {
              " $fragmentRefs": Object {
                "ApplicationDetailsContainer_application": true,
              },
              "applicationRevisionStatus": Object {
                " $fragmentRefs": Object {
                  "ApplicationRevisionStatusContainer_applicationRevisionStatus": true,
                },
              },
              "orderedFormResults": Object {
                "edges": Array [
                  Object {
                    "node": Object {
                      " $fragmentRefs": Object {
                        "ApplicationCommentsContainer_formResult": true,
                      },
                      "id": "abc",
                    },
                  },
                ],
              },
              "rowId": 1,
            },
            "session": Object {
              " $fragmentRefs": Object {
                "defaultLayout_session": true,
              },
            },
          }
        }
        review={true}
      />
      <Relay(IncentiveCalculator)
        query={
          Object {
            " $fragmentRefs": Object {
              "ApplicationDetailsContainer_query": true,
              "IncentiveCalculatorContainer_query": true,
            },
            "application": Object {
              " $fragmentRefs": Object {
                "ApplicationDetailsContainer_application": true,
              },
              "applicationRevisionStatus": Object {
                " $fragmentRefs": Object {
                  "ApplicationRevisionStatusContainer_applicationRevisionStatus": true,
                },
              },
              "orderedFormResults": Object {
                "edges": Array [
                  Object {
                    "node": Object {
                      " $fragmentRefs": Object {
                        "ApplicationCommentsContainer_formResult": true,
                      },
                      "id": "abc",
                    },
                  },
                ],
              },
              "rowId": 1,
            },
            "session": Object {
              " $fragmentRefs": Object {
                "defaultLayout_session": true,
              },
            },
          }
        }
      />
    </Col>
    <Col
      className="application-comments"
      md={4}
    >
      <Relay(ApplicationCommentsComponent)
        formResult={
          Object {
            " $fragmentRefs": Object {
              "ApplicationCommentsContainer_formResult": true,
            },
            "id": "abc",
          }
        }
        key="abc"
        review={true}
      />
    </Col>
  </ForwardRef>
  <JSXStyle
    id="1070550671"
  >
    .container.jsx-1070550671{display:none;}
  </JSXStyle>
</Relay(DefaultLayout)>
`;
